### PR TITLE
Add basic Zuul jobs

### PR DESCRIPTION
**Pull request by mnaser**
❓ **What kind of change does this PR introduce?**
  - [ ] 🐞 bug fix
  - [ ] 🐣 feature
  - [ ] 📋 docs update
  - [ ] 📋 tests/coverage improvement
  - [ ] 📋 refactoring
  - [ ] 💥 other

📋 **What is the related issue number (starting with `#`)**

#

❓ **What is the current behavior?** (You can also link to an open issue here)

❓ **What is the new behavior (if this is a feature change)?**

📋 **Other information**:

📋 **Checklist**:

  - [ ] I think the code is well written
  - [ ] I wrote [good commit messages][1]
  - [ ] I have [squashed related commits together][2] after the changes have been approved
  - [ ] Unit tests for the changes exist
  - [ ] Integration tests for the changes exist (if applicable)
  - [ ] I used the same coding conventions as the rest of the project
  - [ ] The new code doesn't generate linter offenses
  - [ ] Documentation reflects the changes
  - [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

---
This change is [Reviewable](https://reviewable.io/reviews/cherrypy/cheroot/267)

Depends-On: https://review.opendev.org/711056

### CODE REVIEW

**Comment by mnaser** [comment]
This didn't fire because `master` is not actually a protected branch. I'm hoping we can make it a protected branch so that Zuul runs CI on that.

**Comment by mnaser** [comment]
I'm disabling the option to run only against protected branches to workaround so I can progress but I really think that we should work on enabling it.  I'll add an issue for it

**Comment by mnaser** [comment]
recheck

**Comment by mnaser** [comment]
recheck

File: .zuul.yaml [hunk 1]
```diff
@@ -1,4 +1,52 @@
+- job:
+    name: tox-pre-commit
+    parent: tox
+    vars:
+      tox_envlist: pre-commit
+      # NOTE(mnaser): Drop this once zuul/zuul-jobs no longer prefers
+      #               Python 2 by default.
+      tox_prefer_python2: false
```
Comment (webknjaz) [inline .zuul.yaml]: Actually, almost all jobs should run under Python 3. Especially, linters. There should be a matrix of Python versions for pytest tests, and only these should have one Python 2.7 job (until we drop it b/c of EOL).

File: bindep.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+gcc
+libenchant1c2a
+python3-dev
+
```
Comment (webknjaz) [inline bindep.txt]: Could you please drop empty lines?
```suggestion
```

**Comment by webknjaz** [comment]
recheck

**Pull request merged**